src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -2,88 +2,213 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
+using System.Diagnostics;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public static class CalendarHelpers
     {
-        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
+        public static Calendar[] s_calendars = new Calendar[]
+        {
+            new ThaiBuddhistCalendar(),
+            new JapaneseCalendar()
+        };
+
         private static int MinEra(Calendar calendar) => calendar.GetEra(calendar.MinSupportedDateTime);
+        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
 
-        private static int YearOffset(Calendar calendar) => calendar.GetYear(calendar.MinSupportedDateTime) - 1;
+        private static int YearOffset(Calendar calendar)
+        {
+            if (calendar is ThaiBuddhistCalendar)
+            {
+                return 543;
```

Comment:
I would say get rid of YearOffset and instead get the minimum and maximum dates for all eras in the calendar and then you can test with these dates for the negative values. also you can optimize the implementation of MaxYearInEra and MinYearInEra by using the following implementation 

        static DateTime GetMinDateInEra(Calendar cal, int era)
        {
            int [] eras = cal.Eras;

            // Assert here ere is less than eras[0]

            if (eras.Length == 1 || era == eras[eras.Length - 1] || era == 0)
            {
                return cal.MinSupportedDateTime;
            }

            return cal.ToDateTime(1, 1, 1, 0, 0, 0, 0, era);
        }

        static DateTime GetMaxDateInEra(Calendar cal, int era)
        {
            int[] eras = cal.Eras;
            // Assert here ere is less than eras[0]

            if (eras.Length == 1 || era == eras[0] || era == 0)
            {
                return cal.MaxSupportedDateTime;
            }

            return cal.ToDateTime(1, 1, 1, 0, 0, 0, 0, era + 1).AddDays(-1);
        }